docs/kubespray-upgrade-groups.md

```diff
@@ -0,0 +1,44 @@
+# Kubespray upgrade groups
+
+Currently, the default upgrade method of kubespray, is to take all nodes in sequence to avoid any downtime situations, but this takes too much time.
+So, the goal is that we want to constraint the time an upgrade takes as far as we could, so it can be predictable for both operators & users.
+
+The managed upgrade scripts should allow us to :
+
+- Generate a static inventory with different groups of nodes based on their type ( elastisys, worker, postgres, control plane ... etc)
+
+- Upgrade the control planes first
+
+- Upgrade the groups in parallel by taking only one node per group, in sequence
+
+- Pick the order in which nodes to upgrade first to avoid downtime for databases. In case of primary-secondary replication, we should start with node that host the secondaries first
```

Comment:
```suggestion
- Pick the order in which nodes to upgrade first to avoid downtime for databases. In case of primary-secondary replication, we should start with node that host the secondaries first (not yet implemented).
```
Just to make it clear.

Reply:
Actually, the script is getting the nodes which hosts the postgres pods that have the label *spilo-role* equal to master, and append them as the last element in the group, so they will be the last nodes to get upgraded

Reply:
Wha :sweat_smile: I missed that. Is it possible to make that configurable with a list of key=value pairs to look for?

Reply:
Where do you do this check?

Reply:
I think I didn't understand correctly the key=value pair configuration, is it specific to postgres ( ie: spilo-role=master )?

Reply:
I was thinking to make it more generic, so you can define for example `[spilo-role: master, redisfailovers-role: master]` in the config and it would check that if any node contains a pod matching any of the set labels it would just put that last in the group.

Reply:
I don't think it is feasible after some trial, as we're not relying only on those labels. It seems like for each component we need specific labels to determine the nodes of a cluster and then get the ones hosting the masters. 
I've added anyway support for `redisfailovers-role ` in a separate condition, so now nodes hosting redis masters will be the last to get upgraded

Reply:
Not really, we only need to do an early get on all namespaces with the configured labels and get the list of nodes that host those pods, then when we're adding to the inventory we just do a lookup and see if it exists in that list, then we should put it last in it's group.

I'm fine with doing that in a separate PR as this works although it is static.